lua/ttt2/libraries/modelbuilder.lua

```diff
@@ -0,0 +1,316 @@
+---
+-- Used to build custom world and view models. 
+-- @author Mineotopia
+-- @module modelbuilder
+
+---@class ModelData
+---@field type string The type of this model, it can be "Model", "SPrite" or "Quad"
+---@field model string The path to the model file
+---@field bone string The name of the reference bone
+---@field rel string The name of a related identifier
+---@field pos Vector The position offset of the model
+---@field angle Angle The rotation offset of the model
+---@field size Vector The size multiplier of the model
+---@field color Color The color for the model
+---@field surpresslightning boolean Set to true if the engine lightning should be suppressed
+---@field material string The model material
+---@field skin number The skin index of the model
+---@field bodygroup table A table of bodygroups
+
+---@class BoneData
+---@field scale Vector The scale multiplier
+---@field pos Vector The position offset
+---@field angle Angle The rotation offset
+
+if SERVER then
+	AddCSLuaFile()
+
+	return
+end
+
+modelbuilder = {}
```

Comment:
I think we should rename this, because it is much more connected to the weapon base and also mainly takes care of rendering instead of building the models.

Reply:
modelrenderer then?

Reply:
weaponmodelrenderer ? idk lets ask Chief Naming Officer @Histalek

Reply:
do we really need the 'model' part? shouldn't `weaponrenderer` be just fine?

Reply:
that's fine as well, yes